pandas/core/reshape/merge.py

```diff
@@ -1307,15 +1310,20 @@ def _get_join_indexers(
     # factorize keys to a dense i8 space
     # `count` is the num. of unique keys
     # set(lkey) | set(rkey) == range(count)
-    lkey, rkey, count = _factorize_keys(lkey, rkey, sort=sort)
 
+    # flip left and right keys if performing a right merge
```

Comment:
so another possiblity is to pass how= to _factorize_keys, this *maybe* a more complicated code path; pls comment one way or the other (in the PR).

Reply:
@jreback Sure, I've gone with that and it seems a lot simpler